### ISSUE TITLE
allowing eslint-disable-next-line directives to have extra lines only for descriptions

### ISSUE THREAD

**Issue by PixnBits** [eslint/eslint#14284]
**The version of ESLint you are using.**
eslint@7.22.0

**The problem you want to solve.**
With the adoption of the [description in directive comments RFC](https://github.com/eslint/rfcs/blob/master/designs/2019-description-in-directive-comments/README.md) eslint-disable-next-line directives can get long. 
```javascript
// eslint-disable-next-line global-require -- description on why this exception is seen as appropriate but past a comfortable reading line length
const fizz = require(buzz);
```

When using max-len these lines are reported as errors (by [getDirectiveComments](https://github.com/eslint/eslint/blob/9d6063add931f0803cae1676d5df307baf114360/lib/linter/linter.js#L326-L333), à la #10334)
```
/path/to/file.js
  15:2  error  eslint-disable-next-line comment should not span multiple lines
```
This pushes developers to either get too terse and remove possibly important information or to disable and re-enable the rule
```javascript
/* eslint-disable global-require --
   description on why this exception is seen as appropriate but past a 
   comfortable reading line length */
const fizz = require(buzz);
/* eslint-enable global-require -- re-enable */
```
Developers may forget to re-enable the rule, though this may be mitigated by [eslint-comments/disable-enable-pair](https://github.com/mysticatea/eslint-plugin-eslint-comments/blob/master/docs/rules/disable-enable-pair.md). If they use a rule like [eslint-comments/require-description](https://github.com/mysticatea/eslint-plugin-eslint-comments/blob/master/docs/rules/require-description.md) the re-enabling comment may become tedious.

**Your take on the correct solution to problem.**
A pit of success may be allowing eslint-disable-next-line directives to have extra lines but only for the description. e.g.
```javascript
/* eslint-disable-next-line global-require --
   description on why this exception is seen as appropriate but past a 
   comfortable reading line length */
const fizz = require(buzz);
```

**Are you willing to submit a pull request to implement this change?**
I may need guidance as it would be my first in this project, but yes. Eslint has given me a lot over the years, proposing an implementation seems more than fair.

**Comment by nzakas** [comment]
That’s an interesting idea and seems like a reasonable solution to the problem. I’d like to hear what the rest of @eslint/eslint-tsc thinks.

**Comment by btmills** [comment]
I'd be open to this as a "pit of success" as you said. When I've encountered this issue in the past, I've used this style, but the downside is that it could conflict with `multiline-comment-style` for those who enable that rule.

```js
// description on why this exception is seen as appropriate but past a
// comfortable reading line length
// eslint-disable-next-line global-require
const fizz = require(buzz);
```

**Comment by mdjermanovic** [comment]
Makes sense to me, multiline `eslint-disable-next-line`  is probably less confusing than multiline `eslint-disable-line`.

> A pit of success may be allowing eslint-disable-next-line directives to have extra lines but only for the description

As for the reasons described in #10334, where this restriction was introduced, is there a difference between allowing extra lines only for the description and allowing extra lines in general?

**Comment by nzakas** [comment]
Nice research! I didn’t even know we did that.

I think allowing extra lines only for the description, and perhaps only following a character sequence like dash-dash could be a good signifier. I’m not sure I agree with the original premise of #10334, as I think even multi line descriptions could easily fall under the “but which line?” argument.

I’d be fine with reversing #10334 entirely as well.

**Comment by villesau** [comment]
Would be cool if e.g following would be supported `// eslint-disable-next-line react/no-array-index-key there is no other unique identiefier`. For example `// @ts-expect-error works this way and let's you write why this is here`. Alternatively the syntax could be e.g `// eslint-disable-next-line react/no-array-index-key # there is no other unique identiefier` if simple space is not compatible.

**Comment by btmills** [comment]
I'd be okay reversing #10334 for `eslint-disable-next-line` comments to allow multiple lines because it's intuitive to me that the comment would then apply to the next line after the comment ends.

**Comment by snitin315** [comment]
I will work on it.
